### ISSUE TITLE
Document `declaration-property-value-no-unknown` appropriate languages

### ISSUE THREAD

**Issue by crystalfp** [stylelint/stylelint#6638]
### What steps are needed to reproduce the bug?

Using the following `bug.vue` file:
```
<template>
<ul>
    <li>Item</li>
</ul>
</template>

<style lang="scss" scoped>
@use "sass:color";

$spacing: 1.5rem;
$radius: 10px;
$base: red;

li {
  padding-left: calc(2 * $spacing - $radius - 3px);
  background-color: color.scale($base, $lightness: -8%);
}

ul {
  margin-left: calc($radius - $spacing);
  padding-left: 0;
}
</style>
```

### What Stylelint configuration is needed to reproduce the bug?

```yaml
extends:
  - stylelint-config-standard
  - stylelint-config-standard-vue
  - stylelint-config-recommended-vue/scss

overrides:
- files:
  - "*.vue"
  - "**/*.vue"
  customSyntax: postcss-html

rules:
  declaration-property-value-no-unknown: true
```

### How did you run Stylelint?

CLI with stylelint bug.vue and postcss-html

### Which version of Stylelint are you using?

15.0.0

### What did you expect to happen?

No problem reported

### What actually happened?

```
bug.vue
 16:3  ✖  Cannot parse property value "calc(2 * $spacing - $radius - 3px)"   parseError
 17:3  ✖  Cannot parse property value "color.scale($base, $lightness: -8%)"  parseError
 21:3  ✖  Cannot parse property value "calc($radius - $spacing)"             parseError

3 problems (3 errors, 0 warnings)
```

### Does the bug relate to non-standard syntax?

Yes, related to SCSS variables and functions

### Proposal to fix the bug

No suggestions, but if inside the two `calc()` functions I substitute the variables `$radius` and `$spacing` with their values, at these two errors disappear. Also, if I remove the rule `declaration-property-value-no-unknown: true` the error disappears.

**Comment by jeddy3** [comment]
@crystalfp Thanks for the report and for using the template.

I [can reproduce this issue](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKAuEAO8ViQBswAKAEgDMIIBKGlAXwB0A7EAGhAGcBPFjJAA8E4Llw7gILCgEsA5iOCsUKJiABOAV0IwuavEpYqVa2GEJJ1SDDKkBabOpwx1GHnYBuRTTDssIdposANb+AO4s+phaMMqMrAwgDEA) with:

```scss
a { top: calc($foo)) }
```

The built-in rules are geared towards standard CSS syntax. And this particular rule "considers values for properties defined within the CSS specifications to be known". You may need to turn the rule off if you're using non-standard syntax, like SCSS dollar variables.

Having said that, we usually silently ignore non-standard syntax in our rules rather than flag problems so that they remain generally helpful to everyone. We'd typically reach for one of our `isStandardSyntax*` utils to do this.

However, that might not be appropriate here as it'd involve parsing the value twice:

- first with PostCSS (a forgiving parser) to determine if there's a dollar variable in a function
- then with CSSTree to detect syntax problems

Alternatively, we can silently ignore parse errors rather than report them, which would make the rule useful to those using non-standard syntax but less useful to people who aren't.

Ultimately, the [SCSS Stylelint community](https://github.com/stylelint-scss) should create their own version of the rule (like they already do for a number of the built-in ones) that uses CSSTree's [syntax extension](https://www.npmjs.com/package/stylelint-csstree-validator#syntaxextensions).

Until that happens, we can either:

- silently ignore parse errors
- advise people using non-standard syntax not to not use the rule

Does anyone else have any ideas?

**Comment by crystalfp** [comment]
I understand and it's a pity I have to disable it, because this rule discovered two long standing errors in my code.
Anyway, I hope it will reappears in `stylelint-config-recommended-scss`, `stylelint-config-recommended-vue/scss` or `stylelint-config-standard-scss`.

**Comment by ybiquitous** [comment]
@jeddy3 Thanks for deepening into the problem. Your investigation is so helpful.

Indeed, extending a `isStandardSyntax*` util (`isStandardSyntaxValue` in this case) seems hard to find SCSS variables in functions because the util uses just a regex for simple cases, like this:

https://github.com/stylelint/stylelint/blob/c64065e71f654533d973002d4b5cc02e2e3c3a5b/lib/utils/isStandardSyntaxValue.js#L24-L27

We usually prefers false negatives over false positives, so

> silently ignore parse errors

looks reasonable. Though I have a bit concern that we might not find when CSSTree has some bugs.

Because there are many SCSS users and we shouldn't load the SCSS community much, I guess it's better to handle the problem in the built-in rule.

**Comment by jeddy3** [comment]
> Though I have a bit concern that we might not find when CSSTree has some bugs.

Me too. And we'll make the rule less useful for people writing standard CSS.

Additionally, the rule is likely to flag parseable non-standard syntax used by SCSS users, e.g.:

```scss
a { 
  color: adjust-hue(#6b717f, 60deg);
}
```

[(Demo)](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKA6A7FLIBsIAneFJKAKwFcBnAFwFoALKmACgGIA2AIwHYAjHwBmAGhRcADLADmASgDc2AL7YQokDQCeWOkgAeCTWBo114CFmEBLGUeDZcGEESr4YNZ6Qc5cmELBg+EhESHTWlgwADkQQUTBEdFoMAG5I+KwMWBAMVFgA1tkA7lheKHSuMI4oqljKIMpAA)

The SCSS Stylelint community will need to [extend the syntax](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/#propertiessyntax--property-syntax-) when they write their version of the rule.

> Because there are many SCSS users and we shouldn't load the SCSS community much

Most of the other `*-no-unknown` rules have counterparts in [`stylelint-scss`](https://github.com/stylelint-scss/stylelint-scss), and hopefully the SCSS community will create one for this. It's down to the community's health whether it'll happen or not. (More people seem to be writing standard CSS these days, so it may not happen.)

Having thought about it some more, silently ignoring the parse errors feels like we're putting a loose plaster on the problem that could create false expectations for our users.

It's implicitly implied that all the built-in rules are geared towards standard CSS (part of our reorientation in `14.0.0` to supporting just CSS out-of-the-box). We can add a line to this rule's README to make this explicit so that people migrating to `15.0.0` know the score and don't turn on the rule for non-standard syntax.

**Comment by ybiquitous** [comment]
> Most of the other `*-no-unknown` rules have counterparts in [`stylelint-scss`](https://github.com/stylelint-scss/stylelint-scss), and hopefully the SCSS community will create one for this. It's down to the community's health whether it'll happen or not. (More people seem to be writing standard CSS these days, so it may not happen.)

Ah, indeed, that makes sense. 👍🏼 

> We can add a line to this rule's README to make this explicit so that people migrating to `15.0.0` know the score and don't turn on the rule for non-standard syntax.

Agreed. Nice idea. 👍🏼

**Comment by jeddy3** [comment]
> I understand and it's a pity I have to disable it, because this rule discovered two long standing errors in my code.

@crystalfp I'm glad the rule discovered new errors for you. We hope to add similar rules now that we've [stepped back from enforcing stylistic conventions in favour of helping people avoid errors (and enforce non-stylistic consistency)](https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules).

> Anyway, I hope it will reappears in stylelint-config-recommended-scss, stylelint-config-recommended-vue/scss or stylelint-config-standard-scss.

If you've time, I believe the SCSS Stylelint community appreciates help [creating custom rules](https://stylelint.io/developer-guide/plugins) for their plugin.
